torch/distributed/_sharded_tensor/ops/elementwise_ops.py

```diff
@@ -0,0 +1,35 @@
+import copy
+
+import torch
+from torch.distributed._sharded_tensor import (
+    sharded_op_impl,
+    Shard,
+    ShardedTensor,
+)
+
+
+@sharded_op_impl(torch.nn.functional.gelu)
+def gelu(types, args=(), kwargs=None, pg=None):
+    """
+    Handles ``__torch_function__`` dispatch for the elementwise op
+    ``torch.nn.functional.gelu``.
+    This method computes on either a normal tensor or a sharded tensor.
+    """
+    input = args[0]
+    op = torch.nn.functional.gelu
+    # Validate types
+    if not isinstance(input, torch.Tensor) and not isinstance(input, ShardedTensor):
+        raise TypeError("input needs to be either torch.Tensor or ShardedTensor")
+    if isinstance(input, torch.Tensor):
+        return op(args[0])
+    else:
+        local_shards_new = []
+        for local_shard in input.local_shards():
+            local_shards_new.append(Shard(op(local_shard.tensor), local_shard.metadata))
+        new_st = ShardedTensor._init_from_local_shards(
+            local_shards_new, input.size(), process_group=pg
+        )
```

Comment:
This logic is probably common across all elementwise ops. To make this generic, can we support both `gelu` and `relu` and have them both share the same code? In an ideal world, just adding one line of code should be sufficient to add a new elementwise op:

```
register_elementwise_op(torch.nn.functional.gelu)
register_elementwise_op(torch.nn.functional.relu)
```

Reply:
Sure. Let me make this a decorator so that we can easily support `elementwise_op` in a just one line.